provider/openstack/provider.go

```diff
@@ -648,6 +649,11 @@ func (e *Environ) Bootstrap(ctx environs.BootstrapContext, args environs.Bootstr
 func (e *Environ) supportsNeutron() bool {
 	client := e.client()
 	endpointMap := client.EndpointsForRegion(e.cloud.Region)
+	if len(endpointMap) == 0 {
+		// In some corner cases the endpoingMap is empty.  Default to
```

Comment:
the 3 places calling supportsNeutron() try to get an authenticated client if the client isn't already.... looks like something else is going on.  will investigate.  all the goose output of endpoint data i saw did include "network".